Test/Utils/ComputeNode/Service.ps1

```diff
@@ -6,48 +6,52 @@
 class Service {
     [string] $ServiceName;
     [string] $ExecutablePath;
-    [string] $LogPath;
-    [AllowEmptyCollection()] [string[]] $CommandLineParams;
+    [hashtable] $AdditionalParams;
 
-    [void] init([string] $ServiceName, [string] $ExecutablePath, [string] $LogPath, [string[]] $CommandLineParams) {
+    [void] init([string] $ServiceName, [string] $ExecutablePath, [hashtable] $AdditionalParams) {
         $this.ServiceName = $ServiceName
         $this.ExecutablePath = $ExecutablePath
-        $this.LogPath = $LogPath
-        $this.CommandLineParams = $CommandLineParams
+        $this.AdditionalParams = $AdditionalParams
     }
 
-    Service ([string] $ServiceName, [string] $ExecutablePath, [string] $LogPath, [string[]] $CommandLineParams) {
-        $this.init($ServiceName, $ExecutablePath, $LogPath, $CommandLineParams)
+    Service ([string] $ServiceName, [string] $ExecutablePath, [hashtable] $AdditionalParams) {
+        $this.init($ServiceName, $ExecutablePath, $AdditionalParams)
     }
 }
 
 function Install-ServiceWithNSSM {
     Param (
-        [Parameter(Mandatory=$true)] $Session,
-        [Parameter(Mandatory=$true)] $ServiceName,
-        [Parameter(Mandatory=$true)] $ExecutablePath,
-        [Parameter(Mandatory=$false)] [AllowEmptyCollection()] [string[]] $CommandLineParams
+        [Parameter(Mandatory=$true)] [PSSessionT] $Session,
+        [Parameter(Mandatory=$true)] [Service] $Configuration
     )
 
     $Output = Invoke-NativeCommand -Session $Session -ScriptBlock {
-            nssm install $using:ServiceName "$using:ExecutablePath" $using:CommandLineParams
+        nssm install $Using:Configuration.ServiceName "$Using:Configuration.ExecutablePath"
     } -AllowNonZero -CaptureOutput
 
     $NSSMServiceAlreadyCreatedError = 5
     if ($Output.ExitCode -eq 0) {
         Write-Log $Output.Output
     }
     elseif ($Output.ExitCode -eq $NSSMServiceAlreadyCreatedError) {
-        Write-Log "$ServiceName service already created, continuing..."
+        Write-Log "$Configuration.ServiceName service already created, continuing..."
```

Comment:
Otherwise it will resolve $Configuration and concatenate '.Servicename'
```suggestion
        Write-Log "$($Configuration.ServiceName) service already created, continuing..."
```